### PR TITLE
fix: error eslint config in react-srcipts > 1.0.0

### PR DESCRIPTION
**Pull request by OuYancey**
**Break Change**: Need to consider compatibility with react-scripts < 1.0.0.

### CODE REVIEW

File: packages/react-app-rewire-eslint/index.js [hunk 1]
```diff
@@ -1,7 +1,10 @@
 function rewireEslint(config, env) {
-  delete config.eslint.configFile;
-  config.eslint.useEslintrc = true;
-
+  // when `react-scripts` version < 1.0.0, use next two line
+  // delete config.eslint.configFile;
+  // config.eslint.useEslintrc = true;
+  
+  // `react-scripts` >= 1.0.0
+  config.module.rules[0].use[0].options.useEslintrc = true;
```
Comment (Guria) [inline packages/react-app-rewire-eslint/index.js]: try to use getLoader helper from `react-app-rewired` so not to hard code loader position.
Also we can keep support older setup.
```
const eslintMatcher = (rule) => rule.loader === require.resolve('eslint-loader');
const eslintOptions = config.eslint || getLoader(config.module.rules, eslintMatcher);
eslintOptions.useEslintrc = true ;

```

Comment (OuYancey) [reply packages/react-app-rewire-eslint/index.js]: oh, it's really better... thx.

File: packages/react-app-rewire-eslint/package.json [hunk 2]
```diff
@@ -6,5 +6,8 @@
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1"
   },
+  "peerDependencies": {
```
Comment (Guria) [inline packages/react-app-rewire-eslint/package.json]: No need in it now, I suppose.

File: packages/react-app-rewire-eslint/index.js [hunk 3]
```diff
@@ -1,7 +1,27 @@
+const path = require('path');
+const { getLoader } = require('react-app-rewired');
+
+// this is the path of eslint-loader `index.js`
+const ESLINT_PATH = `eslint-loader${path.sep}index.js`;
```
Comment (Guria) [inline packages/react-app-rewire-eslint/index.js]: Why not to use require.resolve?

**Comment by Guria** [comment]
Would be great to see result squashed into one commit

File: packages/react-app-rewire-eslint/package.json (hunk 2 above)
Comment (OuYancey) [reply packages/react-app-rewire-eslint/package.json]: yes, you are right.

File: packages/react-app-rewire-eslint/index.js (hunk 3 above)
Comment (OuYancey) [reply packages/react-app-rewire-eslint/index.js]: use the same style with other packages. 
see: https://github.com/timarney/react-app-rewired/blob/master/packages/react-app-rewire-less/index.js#L8

Comment (Guria) [reply packages/react-app-rewire-eslint/index.js]: ok let's stick to it for now

**Comment by OuYancey** [comment]
sry, i did wrong HEAD in squash. wait moment.

**Comment by timarney** [comment]
@Guria feel free to merge this (whenever) and I'll publish it.

**Comment by Guria** [comment]
@timarney I have no merge button on this repo yet

**Comment by timarney** [comment]
@Guria think we got the no merge button issue sorted out correct?

**Comment by timarney** [comment]
published

**Pull request merged**